pages/explorer/unitBrowser.tsx

```diff
@@ -0,0 +1,91 @@
+import { NextPage } from "next";
+import { ebpsStats, EbpsType, setEbpsStats } from "../../src/unitStats/mappingEbps";
+import { sbpsStats, SbpsType, setSbpsStats } from "../../src/unitStats/mappingSbps";
+import { setWeaponStats, WeaponStats, WeaponType } from "../../src/unitStats/mappingWeapon";
+import {
+  setUpgradesStats,
+  upgradesStats,
+  UpgradesType,
+} from "../../src/unitStats/mappingUpgrades";
+import { setLocstring, unitStatsLocString } from "../../src/unitStats/locstring";
+import Head from "next/head";
+import React from "react";
+import { generateKeywordsString } from "../../src/head-utils";
+import { getMappings } from "../../src/unitStats/mappings";
+import { UnitTable } from "../../components/unitStats/unitTable";
+
+interface SquadProps {
+  weaponData: WeaponType[];
+  sbpsData: SbpsType[];
+  ebpsData: EbpsType[];
+  upgradesData: UpgradesType[];
+  locstring: any;
+}
+
+// Parameter in Curly brackets is destructuring for
+// accessing attributes of Props Structure directly
+const squadPage: NextPage<SquadProps> = ({
+  weaponData,
+  sbpsData,
+  ebpsData,
+  upgradesData,
+  locstring,
+}) => {
+  // Save data again in global varible for clientMode
+  if (!WeaponStats) setWeaponStats(weaponData);
+
+  if (!ebpsStats) setEbpsStats(ebpsData);
+
+  if (!upgradesStats) setUpgradesStats(upgradesData);
+
+  if (!sbpsStats) setSbpsStats(sbpsData);
+
+  if (!unitStatsLocString) setLocstring(locstring);
```

Comment:
Btw what do we need to do this? I don't think something like this is necessary. 
We should try to pass all the data via props to the components.

Reply:
For the unit browser I don't have the patch version mechanic, so I don't need to globaly memorize which data have been already fetched (to avoid double fetching). I removed those lines but I'm not sure if it makes a difference, since we only saved a reference to an existing data instance. We didn't create a clone as passing an object to a setter function should be call-by-reference and not call-by-value. So by memory consumption the outcome should be more or less the same I guess.

Reply:
I want to look into the DPS calculator little bit more to understand why we need it there. But lately I don't have much free time :'(